internal/models/user.go

```diff
@@ -1,16 +1,16 @@
 package models
 
 import (
-	"time"
-
 	"gorm.io/gorm"
 )
 
 type User struct {
-	ID        uint           `gorm:"primarykey" json:"id"`
-	Email     string         `gorm:"uniqueIndex;not null" json:"email"`
-	Password  string         `gorm:"not null" json:"-"`
-	CreatedAt time.Time      `json:"created_at"`
-	UpdatedAt time.Time      `json:"updated_at"`
-	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
+	gorm.Model
+	Email     string  `gorm:"uniqueIndex;not null" json:"email"`
+	Password  *string `json:"-"` // Nullable for OAuth users
+	Provider  string  `gorm:"not null;default:'credentials'" json:"provider"`
+	FirstName string  `json:"firstName"`
+	LastName  string  `json:"lastName"`
+	Image     string  `json:"image,omitempty"`
```

Comment:
Shouldn't these three be stored in a `Profile` object that we link to a user? So one profile can have multiple user, where each user represents a method of authenticating?